Community/Tdarr_Plugin_MC93_Migz1FFMPEG.js

```diff
@@ -259,7 +279,18 @@ const plugin = (file, librarySettings, inputs, otherArguments) => {
       // Check if codec of stream is mjpeg/png, if so then remove this "video" stream.
       // mjpeg/png are usually embedded pictures that can cause havoc with plugins.
       if (file.ffProbeData.streams[i].codec_name === 'mjpeg' || file.ffProbeData.streams[i].codec_name === 'png') {
-        extraArguments += `-map -v:${videoIdx} `;
+        extraArguments += `-map -0:v:${videoIdx} `;
+        response.infoLog += `File has an image in it as stream: ${i}, removing it! \n`;
+      } else { // if not an image stream then read it's height and width
+        try {
+          const v_width = file.ffProbeData.streams[i].width;
+          const v_height = file.ffProbeData.streams[i].height;
+          vf_scale_args = `-vf 'scale_cuda=w=${v_width}:h=${v_height}'`;
+          response.infoLog += `Video Dimensions read as Width: ${v_width} Height: ${v_height} \n`;
```

Comment:
Why is this needed? Scaling to the same video width and height makes no sense to me.

Reply:
Transcoding fails on certain files when using hardware acceleration args, adding the scaling args will transcode the file and also adds some speed boost (not sure why!).

Reply:
Hi @UmamaheshwarN - finally got a chance to test out your updated code. This part seems to cause my transcoding to fail - not sure if i'm missing libraries or other deps but I get:

```
2022-11-02T17:44:27.713Z kcUzZrU-s:Node[StevenPC]:Worker[ln9_anhUt]:[2/3] D:/TDarr_1/Tdarr_Node/node_modules/@ffmpeg-installer/win32-x64/ffmpeg.exe -hwaccel cuvid -hwaccel_device 0 -hwaccel_output_format cuda -i W:/media/8array.md0/TV/<snip>.mkv -map 0 -vf scale_cuda=w=1920:h=1080 -c:V hevc_nvenc -gpu 0 -cq:V 19 -b:v 3011k -minrate 2107k -maxrate 3914k -bufsize 6022k -spatial_aq:V 1 -rc-lookahead:V 32 -c:a copy -c:s copy -max_muxing_queue_size 9999 W:/media/scratch/tdarr/<snip>.mkv
<snip>
2022-11-02T17:44:28.720Z kcUzZrU-s:Node[StevenPC]:Worker[ln9_anhUt]:ffmpeg version N-92722-gf22fcd4483 Copyright (c) 2000-2018 the FFmpeg developers
2022-11-02T17:44:28.720Z built with gcc 8.2.1 (GCC) 20181201
2022-11-02T17:44:28.720Z configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-amf --enable-ffnvcodec --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth --enable-libopenmpt
<snip>
2022-11-02T17:44:28.720Z [h264 @ 0000018ad7dac280] decoder->cvdl->cuvidCreateDecoder(&decoder->decoder, params) failed -> CUDA_ERROR_INVALID_VALUE: invalid argument
2022-11-02T17:44:28.720Z
2022-11-02T17:44:28.720Z [h264 @ 0000018ad7dac280] Failed setup for format cuda: hwaccel initialisation returned error.
2022-11-02T17:44:28.720Z
2022-11-02T17:44:28.720Z [AVFilterGraph @ 0000018ad813e800] No such filter: 'scale_cuda'
2022-11-02T17:44:28.720Z Error reinitializing filters!
2022-11-02T17:44:28.720Z Failed to inject frame into filter network: Invalid argument
2022-11-02T17:44:28.720Z Error while processing the decoded data for stream #0:0
2022-11-02T17:44:28.720Z
2022-11-02T17:44:28.720Z Conversion failed!
```


Let me know your thoughts - if it's a "me" problem of if you'd like me to generate more info for debug. Thanks!